redis/src/main/scala/zio/redis/TestExecutor.scala

```diff
@@ -14,19 +16,31 @@ private[redis] final class TestExecutor private (
   hyperLogLogs: TMap[String, Seq[String]]
 ) extends RedisExecutor.Service {
 
+  val codec = ProtobufCodec
+
   def execute(command: Chunk[RespValue.BulkString]): IO[RedisError, RespValue] =
     for {
       name   <- ZIO.fromOption(command.headOption).orElseFail(ProtocolError("Malformed command."))
       result <- runCommand(name.asString, command.tail).commit
     } yield result
 
+  def execute[In: Input, Out: Output](name: String, in: In): IO[RedisError, Out] = {
+    val zio = for {
+      respOut <- runCommand(name, implicitly[Input[In]].encode(in)(codec)).commit
+      decodedOut <- ZIO.effect(implicitly[Output[Out]].unsafeDecode(respOut))
```

Comment:
As suggested in the previous comment this method should be removed. However, hint for future: whenever you end up invoking `implicitly[something]` like here, consider adding the so called summoner to the typeclass (if you can, as here). For example:

```scala
object Input {
  def apply[A](implicit instance: Input[A]) = instance
}
```

That way, the call site becomes just:

```scala
Input[In].encode(...)
```